File: 107. Binary Tree Level Order Traversal II/Java/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        List<List<Integer>> result = new ArrayList<>();
        queue.add(root);
        while (!queue.isEmpty()) {
            List<Integer> currentList = new ArrayList<>();
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                TreeNode node = queue.remove();
                if (node != null) {
                    currentList.add(node.val);
                    queue.add(node.left);
                    queue.add(node.right);
                }
            }
            if (!currentList.isEmpty()) {
                result.add(0, currentList);
            }
        }
        return result;
    }
}

File: 107. Binary Tree Level Order Traversal II/C++/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode *root) {
        queue<TreeNode *> que;
        vector<vector<int>> result;
        que.push(root);
        while (!que.empty()) {
            vector<int> currentList;
            int size = que.size();
            for (int i = 0; i < size; i++) {
                TreeNode *node = que.front();
                que.pop();
                if (node != nullptr) {
                    currentList.push_back(node->val);
                    que.push(node->left);
                    que.push(node->right);
                }
            }
            if (!currentList.empty()) {
                result.insert(result.begin(), currentList);
            }
        }
        return result;
    }
};